src/payment/types.rs

```diff
@@ -474,3 +477,475 @@ impl StrictDecode for ShortChannelId {
 impl lightning_encoding::Strategy for ShortChannelId {
     type Strategy = lightning_encoding::strategies::AsStrict;
 }
+
+#[derive(Clone, Debug, From, PartialEq, Eq)]
+pub enum AnnouncedNodeAddr {
+    /// An IPv4 address/port on which the peer is listening.
+    IPv4 {
```

Comment:
According to rust API guidelines and to match `InetAddr` naming better to use

```suggestion
    IpV4 {
```